### PR TITLE
fix(api): `tx.gas_price` field

### PR DESCRIPTION
**Pull request by perekopskiy**
## What ❔

Fixes `tx.gas_price` field for legacy and EIP-2930 transactions.

## Why ❔

Follow the [spec](https://ethereum.github.io/execution-apis/api-documentation/)

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: zk_toolbox/crates/zk_supervisor/src/commands/test/prover.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use crate::messages::MSG_PROVER_TEST_SUCCESS;
 
 pub fn run(shell: &Shell) -> anyhow::Result<()> {
     let ecosystem = EcosystemConfig::from_file(shell)?;
-    let _dir_guard = shell.push_dir(&ecosystem.link_to_code.join("prover"));
+    let _dir_guard = shell.push_dir(ecosystem.link_to_code.join("prover"));
```
Comment (mm-zk) [inline zk_toolbox/crates/zk_supervisor/src/commands/test/prover.rs:9]: is this related to this PR?

File: zk_toolbox/crates/zk_supervisor/src/commands/database/reset.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ pub async fn run(shell: &Shell, args: DatabaseCommonArgs) -> anyhow::Result<()>
 
     let dals = get_dals(shell, &args.selected_dals)?;
     for dal in dals {
-        logger::info(&msg_database_loading(MSG_DATABASE_RESET_GERUND, &dal.path));
+        logger::info(msg_database_loading(MSG_DATABASE_RESET_GERUND, &dal.path));
```
Comment (mm-zk) [inline zk_toolbox/crates/zk_supervisor/src/commands/database/reset.rs:29]: is this related to this PR?

**Review by mm-zk — APPROVED** [review]

File: zk_toolbox/crates/zk_supervisor/src/commands/test/prover.rs (hunk 1 above)
Comment (perekopskiy) [reply zk_toolbox/crates/zk_supervisor/src/commands/test/prover.rs:9]: well, lints were broken in a different PR (probably here https://github.com/matter-labs/zksync-era/commit/30e072bd695615b0095c8bdcfd62b77c6b0ae5e6), and it prevents this PR from merging
